### DISCUSSION TITLE
what is the equivalent of "make/source.mk" with cmake?

### DISCUSSION THREAD

**Discussion by P-I-Engineer** [iNavFlight/inav#6416] [Q&A] [answered]
I'm trying to put all this code in from this pull request and can't get it to compile.
it's this pull request: https://github.com/iNavFlight/inav/pull/4969/files

i'm guessing it's because "make/source.mk" is different now.

thanks

**Comment by stronnag** [comment, marked as answer]
If you mean "where do I add a new source file to the build", then `src/main/CMakeLists.txt`

**Comment by P-I-Engineer** [comment]
oh yeah! that's it. thank you

**Comment by stronnag** [comment]
Now documented in the developer cmake document.

**Comment by P-I-Engineer** [comment]
thanks, i have very little programming/compiling knowledge, just google/youtube/asking questions where i can.
